client/deposits/list/index.tsx

```diff
@@ -213,6 +213,7 @@ export const DepositsList = (): JSX.Element => {
 		setIsDownloading( true );
 		const downloadType = totalRows > rows.length ? 'endpoint' : 'browser';
 		const userEmail = wcpaySettings.currentUserEmail;
+		const locale = wcpaySettings.locale;
```

Comment:
Just passing by and had some questions:
- Should we give the merchants the choice to toggle this functionality somehow? It would be unexpected for merchants with a different site language to start getting their reports in a different language without any change on their side. So this should be an opt-in IMO.
- When localization is enabled, should we just translate the CSV headers? I believe the email body should also be translated in that case.
- Finally, how do we ensure that the translations in all the languages that we support as site languages work well for these headers? Should there be a whitelist or do we know that these would all be ok?

Reply:
All very good questions.

> Should we give the merchants the choice to toggle this functionality somehow? It would be unexpected for merchants with a different site language to start getting their reports in a different language without any change on their side. So this should be an opt-in IMO.

I think that's a fair observation. Our changes will default all merchants to CSV reports in their site language whether they want it or not. It's not unreasonable to imagine a set of merchants that are happy receiving the reports in English irrespective of their site language.

One way to fix this is to keep the locale for reports as English and add a checkbox on WooPayments page to let merchants optionally receive CSVs in the site language.

Copying @souravdebnath1986 and @rogermattic for their expert thoughts on this discussion.

> When localization is enabled, should we just translate the CSV headers? I believe the email body should also be translated in that case.

I had this exact thought. I also believe we should apply localization to emails too. But I'm not sure if doing so would be within the scope of this issue. IMO, it depends on whether it's just the download report email or even the other email templates that are currently not localized. In the latter case, it'd be a separate project-like effort. Nevertheless, you've brought up a good point.

> Finally, how do we ensure that the translations in all the languages that we support as site languages work well for these headers? Should there be a whitelist or do we know that these would all be ok?

From what I know, the translations are a community effort and there's no way to be sure about 100% coverage of all strings in WooPayments. The good news is that all the top languages are 98% or more translated.

https://translate.wordpress.com/projects/woocommerce/woocommerce-payments

Reply:
Thanks for the quick response, I trust that you can take the conversation forward but happy to assist in case any assistance is needed from our end (I worked on the initial CSV export a bit so have some context).

Reply:
Great questions!

>Should we give the merchants the choice to toggle this functionality somehow? It would be unexpected for merchants with a different site language to start getting their reports in a different language without any change on their side. So this should be an opt-in IMO.

The existing functionality for CSV exports from the client consistently translates content to the site language when the total rows are fewer than the page size. To maintain uniform behavior, it is advisable to extend this translation requirement to server exports. Considering that the site is already translated, selecting an export language for client exports may not be possible.

If the goal is to introduce the language option exclusively for server exports, a user-friendly approach could be converting the Download button into a dropdown. This dropdown can then present two buttons: 'Export in English' and 'Export in site language.' 

>When localization is enabled, should we just translate the CSV headers? I believe the email body should also be translated in that case.

In alignment with Anurag's statement, I believe that translating the email content should not be considered as part of the current issue.

Reply:
Hey! Thanks for pinging me here :) 

I agree with @anu-rock:

> 
>  It's not unreasonable to imagine a set of merchants that are happy receiving the reports in English irrespective of their site language. 
>  One way to fix this is to keep the locale for reports as English and add a checkbox on WooPayments page to let merchants optionally receive CSVs in the site language.

We can add this to our Payments settings.  

> If the goal is to introduce the language option exclusively for server exports, a user-friendly approach could be converting the Download button into a dropdown. This dropdown can then present two buttons: 'Export in English' and 'Export in site language.'

@mordeth I think this could be a nice approach but it might be a bit annoying for the user if they need to select an option each time they click **Download** (which should be a quick, direct action). 
However, this might work well for the first time download. We could display a popup with two language options and a note `Don't worry you can always change it later in the settings`. In this way we can _kill two birds with one stone_, making sure that user downloads the file in the language they wanted and knows where to change it if they need to.

Reply:
So this is how we can do it:

When the user clicks on the Download button, we will show this popup:
<img width="1310" alt="Screenshot 2024-01-09 at 16 52 28" src="https://github.com/Automattic/woocommerce-payments/assets/94967682/ce451e2a-493e-49e5-b8e2-4503fd6f17b3">


And then we should also add this to our Payments Settings:
<img width="1162" alt="Screenshot 2024-01-09 at 16 52 38" src="https://github.com/Automattic/woocommerce-payments/assets/94967682/dd2e965c-0bd5-4143-afe1-23418f524c39">

Please feel free to let me know your feedback!